compiler/modules/front.syntax/parse.bal

```diff
@@ -38,27 +36,89 @@ function createSourceFile(FilePath path, string[] lines) returns SourceFile {
     return new(path, scanLines(lines));
 }
 
+function parseImportDecls(Tokenizer tok) returns ImportDecl[]|err:Syntax {
+    ImportDecl[] imports = [];
+    ImportDecl? im = check parseImportDecl(tok);
+    while im is ImportDecl {
+        imports.push(im);
+        im = check parseImportDecl(tok);
+    }
+    return imports;
+}
+
 function parseImportDecl(Tokenizer tok) returns ImportDecl?|err:Syntax {
     Token? t = tok.current();
     if t != "import" {
-        return;
+        return ();
     }
     Position pos = tok.currentPos();
-
     check tok.advance();
-    t = tok.current();
-    if t is [IDENTIFIER, string] { 
-        string org = t[1];
+    var [org, moduleName] = check parseImportOrgModule(tok);
```

Comment:
I still don't think this is split up as clearly as it can be. We want the functions to relate clearly to the grammar.

I would get rid of `parseImportOrgModule` and `parseImportName` and instead have a function `parseImportNamesRest` that parses

```
("." mod-name)*
```

and returns `string[]`.

Reply:
fixed